### PR TITLE
feat: use runtime metric conventions; add runtime attributes

### PR DESCRIPTION
**Pull request by mattjbray**
See https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/runtime-environment-metrics.md#runtime-environment-specific-metrics---processruntimeenvironment

### CODE REVIEW

File: src/opentelemetry.ml [hunk 1]
```diff
@@ -669,6 +702,15 @@ end
 
     These metrics are emitted after each GC collection. *)
 module GC_metrics  = struct
+  (** See https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/semantic_conventions/process.md#process-runtimes *)
+  let runtime_attributes : Proto.Common.key_value list =
+    Conventions.Attributes.[
+        (Process.Runtime.name, "ocaml");
+        (Process.Runtime.version, Sys.ocaml_version);
+    ]
```
Comment (mattjbray) [inline src/opentelemetry.ml:710]: Question: should we include these in all metrics/traces via `global_attributes`?

Comment (c-cube) [reply src/opentelemetry.ml:710]: sounds useful, but can we do it at the _batch_ level? seems super wasteful to emit these for each span/metrics

**Comment by c-cube** [comment]
I do like it, except I still want to be able to set the service name from within the program :)

**Comment by mattjbray** [comment]
> I do like it, except I still want to be able to set the service name from within the program :)

This PR doesn't change anything about the service name (are you thinking of #12?)

**Pull request merged**